lib/python/picongpu/plugins/plot_mpl/slice_emittance_waterfall_visualizer.py

```diff
@@ -0,0 +1,173 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017-2018 PIConGPU contributors
+Authors: Sophie Rudat, Sebastian Starke
+License: GPLv3+
+"""
+
+from picongpu.plugins.data import EmittanceData
+from picongpu.plugins.plot_mpl.base_visualizer import Visualizer as\
+    BaseVisualizer
+import numpy as np
+from matplotlib.colors import LogNorm
+import matplotlib.pyplot as plt
+
+
+class Visualizer(BaseVisualizer):
+    """
+    Class for creation of waterfall plot with the slice emittance value
+    for each y_slice (x-axis) and iteration (y-axis).
+    """
+
+    def __init__(self, run_directories=None, ax=None):
+        """
+        Parameters
+        ----------
+        run_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        super().__init__(EmittanceData, run_directories, ax)
+        self.cbar = None
+
+    def _create_plt_obj(self, idx):
+        """
+        Implementation of base class function.
+        Turns 'self.plt_obj' into a matplotlib.pyplot.plot object.
+        """
+        slice_emit, y_slices, all_iterations, dt = self.data[idx]
+        np_data = np.zeros((len(y_slices), len(all_iterations)))
+        for index, ts in enumerate(all_iterations):
+            np_data[:, index] = slice_emit[ts][1:]
+        ps = 1.e12  # for conversion from s to ps
+        max_iter = max(all_iterations * dt * ps)
+        # np_data.T * 1.e6 converts emittance to pi mm mrad,
+        # y_slices * 1.e6 converts y slice position to micrometer
+        self.plt_obj[idx] = self.ax.imshow(np_data.T * 1.e6, aspect="auto",
+                                           norm=LogNorm(), origin="lower",
+                                           vmin=1e-1, vmax=1e2,
+                                           extent=(0, max(y_slices*1.e6),
+                                                   0, max_iter))
+        if self.cur_iteration:
+            self.plt_lin = self.ax.axhline(self.cur_iteration * dt * ps,
+                                           color='#FF6600')
+        self.cbar = plt.colorbar(self.plt_obj[idx], ax=self.ax)
```

Comment:
Problems might occur since only one colorbar for possibly multiple imshow objects.